### PR TITLE
Added core functionalities with html & JS alone.

### PR DESCRIPTION
**Pull request by AbelG101**
## Changes
- Added an array of objects to keep the list of books.
- Added a function to add a new book to the collection, with title and author.
- Added a function to remove a book from the collection.
- Added a function to store and retrieve data on and from the local storage.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const addBtn = document.querySelector('.add-btn');
+const booksContainer = document.querySelector('.books-container');
+let books = [];
+
+// retrive books from local storage (if they exist) when page loads
```
Comment (tufoinnkuo10) [inline script.js]: - [ ] Great job.. everything works as expected. kindly remove comments for clean code practices

File: script.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  div.setAttribute('class', 'book');
+  div.appendChild(titleParag);
+  div.appendChild(authorParag);
+  div.appendChild(btn);
+  div.appendChild(hr);
+  booksContainer.appendChild(div);
+}
+window.onload = () => {
+  books = JSON.parse(localStorage.getItem('All Books: '));
+  // if books exist on local storage
```
Comment (tufoinnkuo10) [inline script.js]: - [ ]please remove comments so as to maintain clean code

File: script.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+}
+function removeBook(index) {
+  const booksDiv = document.querySelectorAll('.book');
+  books.splice(index, 1);
+  booksDiv[index].remove();
+  // update the  local storage after the array has been modified
+  localStorage.setItem('All Books: ', JSON.stringify(books));
+}
+document.addEventListener('click', (e) => {
+  // if the btn clicked is not remove-btn then exit
```
Comment (tufoinnkuo10) [inline script.js]: - [ ] please remove this comment too

File: script.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    author: String,
+  };
+  book.title = title.value;
+  book.author = author.value;
+  saveOnLocalStorage(book);
+  addBookOnPage(book);
+}
+function removeBook(index) {
+  const booksDiv = document.querySelectorAll('.book');
+  books.splice(index, 1);
```
Comment (tufoinnkuo10) [inline script.js:49]: - [ optional] always favor the use array methods such as filter instead of splice so it doesn't mutate the array

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Team

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ Add/Remove functions work as expected

Required Changes ♻️
Check the comments under the review

tufoin


Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: script.js (hunk 1 above)
Comment (AbelG101) [reply script.js]: @tufoinnkuo10 I don't understand, since when did it become a bad practice to write descriptive comments on your code?

**Review by codecaiine — APPROVED** [review]
Hi @AbelG101  @SurafelGetachewT , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good use of gitflow :heavy_check_mark: 
- Descriptive pull request :heavy_check_mark: 
- Good commits history :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**